### PR TITLE
Haar 3306 approved premises template updates

### PR DESCRIPTION
**Pull request by indyv-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/DateConversionHelper.kt [hunk 1]
```diff
@@ -20,9 +20,9 @@ class DateConversionHelper {
         "dd/MM/yyyy",
         "dd MMMM yyyy",
       ),
-      // "2024-05-01T12:34:56[.1|12|123|1234|12345|123456][Z][+00:00]"
+      // "2024-05-01T12:34:56[.1|12|123|1234|12345|123456|1234567|12345678|123456789][Z][+00:00]"
       DateConversion(
-        "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,6})?Z?(\\+\\d{2}:\\d{2})?$".toRegex(),
+        "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,9})?Z?(\\+\\d{2}:\\d{2})?$".toRegex(),
```
Comment (indyv-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/DateConversionHelper.kt:25]: Needed this change due to one of the date values received for approved premises

File: src/main/resources/templates/template_hmpps-approved-premises-api.mustache [hunk 2]
```diff
@@ -744,10 +775,281 @@
...
+
     {{/ShortTermAccommodation}}
-    {{^ShortTermAccommodation}}
+
+    {{#ApprovedPremises}}
+    <h2>Approved Premises</h2>
+
+        <h3>Booking Extensions</h3>
+        {{#BookingExtensions}}
+            <table class="summary-list">{{#each .}}<tr><td>{{ @key }}</td><td>{{ this }}</td></tr>{{/each}}</table><br/>
```
Comment (indyv-moj) [inline src/main/resources/templates/template_hmpps-approved-premises-api.mustache:865]: will just output the data in a table using key value pairs for now as we don't know what the format is, tried to use partials to reduce repetition in sections below but couldn't quite get that to work

File: src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/DateConversionHelper.kt (hunk 1 above)
Comment (DaveLlewellynMoJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/DateConversionHelper.kt:25]: is this the milliseconds/nano part of the date time? Regex hurts my brain at the best of times

**Review by DaveLlewellynMoJ — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/DateConversionHelper.kt (hunk 1 above)
Comment (indyv-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/DateConversionHelper.kt:25]: yep, increasing to 9 digits allowed for that but

**Pull request merged**